python/cudf/cudf/core/buffer.py

```diff
@@ -176,30 +205,43 @@ def __init__(
             if isinstance(buf, rmm.DeviceBuffer):
                 self._ptr = buf.ptr
                 self._size = buf.size
-                self._owner = buf
-                return
-            iface = getattr(buf, "__cuda_array_interface__", None)
-            if iface:
-                ptr, size = get_ptr_and_size(iface)
-                self._ptr = ptr
-                self._size = size
-                self._owner = buf
-                return
-            ptr, size = get_ptr_and_size(np.asarray(buf).__array_interface__)
-            buf = rmm.DeviceBuffer(ptr=ptr, size=size)
-            self._ptr = buf.ptr
-            self._size = buf.size
+            else:
+                iface = getattr(buf, "__cuda_array_interface__", None)
+                if iface:
+                    ptr, size = get_ptr_and_size(iface)
+                    self._ptr = ptr
+                    self._size = size
+                else:
+                    ptr, size = get_ptr_and_size(
+                        np.asarray(buf).__array_interface__
+                    )
+                    buf = rmm.DeviceBuffer(ptr=ptr, size=size)
+                    self._ptr = buf.ptr
+                    self._size = buf.size
             self._owner = buf
 
+        if global_manager.enabled:
+            global_manager.get().add_other(self)
```

Comment:
Under what circumstances can we have a mix of `Buffer` and `SpillableBuffer` objects? Is this branch only reachable in the case (now removed) where we create a `Buffer` pointing into a region of memory owned by a `SpillableBuffer`? Otherwise I would expect that in any process we either have all `Buffer` or all `SpillableBuffer` objects based on the behavior of [`as_device_buffer_like` here](https://github.com/rapidsai/cudf/pull/11553/files#diff-32f1c821a9d6a7df78ee27096b55d3527f4e82a28b6ff424400e12f67aa0cfd7R147).

Reply:
Removed "others" in b58c6d01fe15691c8b2c1c08c2fddd3ea2ed5cc9